p2p/module.go

```diff
@@ -199,3 +217,99 @@ func (m *p2pModule) handleNetworkMessage(networkMsgData []byte) {
 
 	m.GetBus().PublishEventToBus(&event)
 }
+
+func configureBootstrapNodes(p2pCfg *configs.P2PConfig, m *p2pModule) error {
+	var csvReader *csv.Reader
+	customBootstrapNodesCsv := strings.Trim(p2pCfg.BootstrapNodesCsv, " ")
+	if customBootstrapNodesCsv == "" {
+		csvReader = csv.NewReader(strings.NewReader(defaults.DefaultP2PBootstrapNodesCsv))
+	} else {
+		csvReader = csv.NewReader(strings.NewReader(customBootstrapNodesCsv))
+	}
+	bootStrapNodes, err := csvReader.Read()
+	if err != nil {
+		return fmt.Errorf("error parsing bootstrap nodes: %w", err)
+	}
+	for _, node := range bootStrapNodes {
+		if !isValidHostnamePort(node) {
+			return fmt.Errorf("invalid bootstrap node: %s", node)
+		}
+	}
+	m.bootstrapNodes = bootStrapNodes
+	return nil
+}
+
+func isValidHostnamePort(str string) bool {
+	pattern := regexp.MustCompile(`^(https?)://([a-zA-Z0-9.-]+):(\d{1,5})$`)
+	matches := pattern.FindStringSubmatch(str)
+	if len(matches) != 4 {
+		return false
+	}
+	protocol := matches[1]
+	if protocol != "http" && protocol != "https" {
+		return false
+	}
+	port, err := strconv.Atoi(matches[3])
+	if err != nil || port < 0 || port > 65535 {
+		return false
+	}
+	return true
+}
+
+func isSelfInAddrBook(selfAddr cryptoPocket.Address, addrBook typesP2P.AddrBook) bool {
+	for _, peer := range addrBook {
+		if peer.Address.Equals(selfAddr) {
+			return true
+		}
+	}
+	return false
+}
+
+func bootstrap(m *p2pModule) error {
+	var (
+		addrBook typesP2P.AddrBook
+	)
+
+	for _, bootstrapNode := range m.bootstrapNodes {
+		m.logger.Info().Str("endpoint", bootstrapNode).Msg("Attempting to bootstrap from bootstrap node") // TODO: deblasis - fix this. For some reason it's not logging
```

Comment:
The logger should already be initialized. Try using the global logger to see if it's a general logger thing or a module-specific logger ting